### PR TITLE
Create script to grab valuesets and store them in CSV files for TTC Model

### PR DESCRIPTION
**Pull request by BradySkylight**
## Description

Created a script to load the various valuesets required for the eICR fields that will be processed in the Text-to-Code solution for the MVP.  This script calls the various APIs necessary to gather the code & text/terms for the specified valuesets and stores them in respective csv files in our project.

### ValueSets & Fields:
* Lab Orders - Loinc - READY (for later use)
* Lab Observations - Loinc - READY (for later use)
* Lab Names (Both Ordering and Resulting) - Loinc - READY
* Lab Result Values - Snomed (Lab Results) - READY
* Lab Result Interpretations - HL7 Lab Interpretations - READY

## Related Issues

[GitHubIssue-30](https://app.zenhub.com/workspaces/dibbs-text-to-code-68756bf8ea06bb00319391c0/issues/gh/cdcgov/dibbs-text-to-code/30)

## Additional Notes

The LOINC and SNOMED (UMLS) APIs require Auth. - I decided to use environment variables to pull these values.  We can update this later to use GitHub secrets if we would like.  In the interim, if anyone on the team needs a username/pwd or API Key, please reach out to me and I can provide them.  

I am including the resulting csv files in this PR as well so they can be leverage in our current work efforts.  If we want to update these files, we can do so at that time by running the script again.

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [X] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [X] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [X] I have updated the documentation, if applicable.
- [X] I have added or updated test cases to cover my changes, if applicable.
- [X] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: scripts/value_sets/terminology_valueset_sync.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+LOINC_LAB_RESULT_SUFFIX = "query=orderobs:Observation+OR+orderobs:Both&count=500"
+HL7_LAB_INTERP_URL = (
+    "https://www.fhir.org/guides/stats2/valueset-us.nlm.vsac-2.16.840.1.113883.1.11.78.json"
+)
+UMLS_SNOMED_LAB_VALUES_URL = (
+    "https://uts-ws.nlm.nih.gov/rest/content/current/source/SNOMEDCT_US/362981000/descendants"
+)
+
+# Get Terminology Usernames and Passwords
+LOINC_USERNAME = os.environ.get("LOINC_USERNAME")
```
Comment (ericbuckley) [inline scripts/value_sets/terminology_valueset_sync.py:18]: I think we should document these in the toplevel docstring, help text if we go that route and raise an appropriate error if they are undefined.

File: scripts/value_sets/terminology_valueset_sync.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    None
+
+
+def get_hl7_lab_interp():  # noqa: D103
+    hl7_filename = "hl7_lab_interp.csv"
+    hl7_response = requests.get(HL7_LAB_INTERP_URL)
+
+    if hl7_response.status_code != 200:
+        print(
+            f"ERROR Retrieving HL7 LAB Interpretation CODES: {hl7_response.status_code}: {hl7_response.text}"
```
Comment (ericbuckley) [inline scripts/value_sets/terminology_valueset_sync.py:37]:
```suggestion
            f"ERROR Retrieving HL7 LAB Interpretation CODES: {hl7_response.status_code}: {hl7_response.text}", file=sys.stderr
```

File: scripts/value_sets/terminology_valueset_sync.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        )
+        return
```
Comment (ericbuckley) [inline scripts/value_sets/terminology_valueset_sync.py:39]: If you want to kill the script at this point, you can just call `sys.exit(1)`

File: scripts/value_sets/terminology_valueset_sync.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+# CSV file settings
+CSV_DIRECTORY = "tmp/"
+
+
+def get_umls_snomed_lab_values():  # noqa: D103
+    if UMLS_API_KEY is None:
+        raise KeyError("UMLS_API_KEY Environment Variable must be set to a proper UMLS API Key!")
+    snomed_filename = "snomed_lab_value.csv"
+    page_num = 1
+    params = {"apiKey": UMLS_API_KEY, "pageNumber": page_num}
```
[the remark this answers is not in the document]
Comment (BradySkylight) [reply scripts/value_sets/terminology_valueset_sync.py:56]: I was trying to see if there was a way to increase that.  Thanks for the tip... I'll fix this for sure.  Nice!

File: scripts/terminology_valueset_sync.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        for result in umls_results:
+            snomed_code = result.get("ui")
+            snomed_text = result.get("name")
+            if snomed_code and snomed_text:
+                result_row = {"code": snomed_code, "text": snomed_text}
+                snomed_rows.append(result_row)
+                snomed_row_count += 1
+
+        page_num += 1
+        params = {"apiKey": UMLS_API_KEY, "pageNumber": page_num}
```
Comment (ericbuckley) [inline scripts/terminology_valueset_sync.py:77]: You're missing the `pageSize` here, alternatively you could also do this
```suggestion
        params['pageNumber'] += 1
```

Comment (BradySkylight) [reply scripts/terminology_valueset_sync.py:77]: Ah, good catch, which is weird that you would have to specify the pagesize each time, but I"ll add it here.  Thanks!!
